k8s/upgrades/0.9.0-0.1.0/claim_blockdevices.sh

```diff
@@ -0,0 +1,69 @@
+#!/bin/bash
+
+############################################################################
+# STEP: Get all block devices present on the cluster and corresponding disk#
+#                                                                          #
+#                                                                          #
+# NOTES: Obtain the pool deployments to perform upgrade operatio           #
+############################################################################
+
+function get_csp_list() {
+    local csp_list=""
+    local spc_name=$1
+    csp_list=$(kubectl get csp -l openebs.io/storage-pool-claim=$spc_name \
+             -o jsonpath="{range .items[*]}{@.metadata.name}:{end}")
+    rc=$?
+    if [ $rc -ne 0 ]; then
+        echo "Failed to get csp related to spc $spc"
+        exit 1
+    fi
+    echo $csp_list
+}
+
+function create_bdc_claim_bd() {
+    local spc_name=$1
+    local disk_name=$2
+    local bd_name=$(echo $disk_name | sed 's|disk|blockdevice|')
+    local bd_details=""
+    local bd_details=$(kubectl get bd $bd_name -n $ns \
+                         -o jsonpath='{.metadata.labels.kubernetes\.io/hostname}:bdc-{.metadata.uid}:{.metadata.namespace}')
+    local node_name=$(echo $bd_details | cut -d ":" -f 1)
+    local bdc_name=$(echo $bd_details | cut -d ":" -f 2)
+    local bdc_namespace=$(echo $bd_details | cut -d ":" -f 3)
+
+     local spc_uid=$(kubectl get spc $spc -o jsonpath='{.metadata.uid}')
+
+    sed "s|@spc_name@|$spc_name|g" bdc-create.tpl.json | sed "s|@bdc_name@|$bdc_name|g" | sed "s|@bdc_namespace@|$bdc_namespace|g" | sed "s|@spc_uid@|$spc_uid|g" | sed "s|@bd_name@|$bd_name|g" | sed "s|@node_name@|$node_name" > bdc-create.json
```

Comment:
Don't we need the `g` suffix in the last `sed` command. Are we sure that `@node_name@` is present only once in the file?

Reply:
added